internal/pkg/mapper/configmetadata/remote_save.go

```diff
@@ -0,0 +1,28 @@
+package configmetadata
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+)
+
+func (m *configMetadataMapper) MapBeforeRemoteSave(recipe *model.RemoteSaveRecipe) error {
+	if !recipe.ChangedFields.Has(`metadata`) {
+		return nil
+	}
+
+	config, ok := recipe.Object.(*model.Config)
+	if !ok {
+		return nil
+	}
+
+	api, err := m.StorageApi()
+	if err != nil {
+		return err
+	}
+	req := api.UpdateConfigMetadataRequest(config).Send()
+	if req.Response.Err() != nil {
+		return req.Response.Err()
+	}
```

Comment:
Tu je bohuzial rovnaky problem, ako sme riesili minule.
Predstav si, ze mas nejaku velku `template`, ktora ma 100 konfiguracii, ulozenie metadat kazdej trva napr. `200ms`.
Na dokoncenie operacie by si cakal `20s`.

Reply:
Ak sa to spusti paralelne, tak, povedzme ze `32` requrestov bezi naraz, tak to dobehne za `~1 sekundu`.

Reply:
IMHO by sa to malo riesit v `Scheduler`: https://github.com/keboola/keboola-as-code/blob/3f297d4ed1feb6c615d5c922f8adeeb2bc5feb14/internal/pkg/mapper/scheduler/remote.go#L25-L33

Reply:
Tj. najprv sa ulozia vsetky configs, potom sa ulozia vsetky metadata.
Teraz je to naopak, co by samo o sebe nemuselo vadit, ... problem je, ze to nie je paralelne.

Reply:
Dava to zmysel?

Reply:
No jasně, to mi taky nepřišlo ok. Ale ten MapBeforeRemoteSave se volá pro každý konfig zvlášť a navíc jsi nechtěl aby měl mapper state. Kam bych měl ten pool uložit a kde by se měl potom zavolat Send()?

Reply:
Jo `OnRemoteChange`, ok, už rozumím. 🙈 Akorát mi není jasný proč to dávat do scheduler mapperu. 🤔

Reply:
Pozri ten `Sheduler`, tam je to pre vsetky objekty naraz. Mame 2 metody `MapBeforeRemoteSave/OnRemoteChange` , prva je per object, ... druha pre vsetky.

Snazil som sa to vysvetlit v komentaroch v `mapper.go`, ak to nestaci, tak to skus prosim nejak doplnit, aby to bolo jasne.

Reply:
@JakubMatejka nie do `scheduler`, asi som to blbo napisal ... to je len priklad, ze TAKTO je to super spravit.
Nechas to tu, kde to je.

Reply:
Viac menej rovnako, len nad Storage API a iny api call.

Tam to je tak, ze "ked ulozis (scheduler) config, tak ho aktivuj v Scheduler API"
A ty budes mat: "ked ulozis config, tak uloz jeho metadata cez Storage API"

Reply:
Ano, tam mi chyba jedno slovo :) `IMHO by sa to malo riesit v` -> IMHO by sa to malo riesit TAK AKO v`

Reply:
Jo tak. 😃  Psal jsi `IMHO by sa to malo riesit v Scheduler` tak mě to celý nějak zmátlo. No u těch event mapperů pořád ještě trochu tápu. Většinou se dívám do ostatních mapperů a ten `OnRemoteChange` mi unikl. Imho by bylo fajn kdyby k tomu bylo nějaký schéma kdy se zavolá která eventa při spuštění commandu, už jsem se to několikrát pokoušel doklikat v IDE ale moc se mi to nepodařilo. (koukám že tady už je to i napsaný https://keboola.atlassian.net/wiki/spaces/KB/pages/2380398593/Keboola+as+Code+-+Dev+docs+-+proposal 🙂)

Reply:
Jo a ještě mě tam matou trochu ty názvy. `OnRemoteChange` chápu tak že "když se v remote (tj. v API) něco změní" ale reálně to znamená že ta eventa na základě lokálních změn jde něco měnit do API, tak by se možná měly jmenovat spíš naopak. 😄

Reply:
Vymyslime na calle ;)